test/suite/test_compact07.py

```diff
@@ -0,0 +1,222 @@
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+            
+        # Create a small table that compact should skip over.
+        uri_small = self.uri_prefix + '_small'
+        ds = SimpleDataSet(self, uri_small, self.table_numkv // 2, key_format=self.key_format, value_format=self.value_format)
+        ds.populate()
+        self.delete_range(uri_small)
+              
+        # Create n tables for background compaction to loop through.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            ds = SimpleDataSet(self, uri, self.table_numkv, 
+                            key_format=self.key_format, 
+                            value_format=self.value_format)
+            ds.populate()
+    
+            # Now let's delete a lot of data ranges. Create enough space so that compact runs in more
+            # than one iteration.
+            self.delete_range(uri)
+
+        # Reopen the connection to force the object to disk.
+        self.reopen_conn()
+        
+        small_file_free_space = self.get_free_space(uri_small)
+        
+        # Allow background compaction to run for some time. Set a low free_space_target larger
+        # than the smaller file.
+        self.session.compact(None,f'background=true,free_space_target={small_file_free_space + 1}MB')
+        
+        # Check that the background server is running
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        self.assertEqual(compact_running, 1)
+        stat_cursor.close()
+        
+        # Background compaction should run through every file as listed in the metadata file.
+        # Periodically check how many files we've compacted until we compact all of them.
+        while self.get_files_compacted() < self.n_tables:
+            time.sleep(2)
+            
+        # Check that we made no progress on the small file.
+        self.assertEqual(self.get_pages_rewritten(self.uri_prefix + '_small'), 0)
+        
+        # Check the background compaction server stats. We should have skipped at least once and 
+        # been successful at least once.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        skipped = stat_cursor[stat.conn.background_compact_skipped][2]
+        success = stat_cursor[stat.conn.background_compact_success][2]
+        self.assertGreater(skipped, 0)
+        self.assertGreater(success, 0)
+        stat_cursor.close()
+        
+        # Open a new session for foreground compaction, while leaving the background
+        # compaction server running.
+        session2 = self.conn.open_session()
+        # Use a free_space_target that is guaranteed to run on the small file.
+        session2.compact(uri_small,f'free_space_target={small_file_free_space - 1}MB')
+
+        time.sleep(1)
```

Comment:
why do we need to sleep here?

Reply:
We don't need one, removed in f2f6aef3257e66e47c099483d7f6d82d737a70bf. Thanks for catching this.